tuf/ngclient/updater.py

```diff
@@ -129,6 +130,23 @@ def refresh(self) -> None:
             DownloadError: Download of a metadata file failed in some way
         """
 
+        if self.config.offline:
+            try:
+                # Try loading only local data
+                data = self._load_local_metadata(Timestamp.type)
+                self._trusted_set.update_timestamp(data, offline=True)
+                data = self._load_local_metadata(Snapshot.type)
+                self._trusted_set.update_snapshot(data, trusted=True, offline=True)
+                data = self._load_local_metadata(Targets.type)
+                self._trusted_set.update_delegated_targets(
+                    data, Targets.type, Root.type, offline=True
+                )
```

Comment:
Looking at all of the calls taking the same argument, we should probably consider "offline" as an attribute of the TrustedMetadataSet -- a constructor argument could work.